### PR TITLE
grpc context propagation with task name

### PR DESCRIPTION
**Pull request by danielnorberg**
Make outer grpc context and task name available to logging etc in user code (process fn).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -109,6 +109,11 @@
         <artifactId>kryo-shaded</artifactId>
         <version>4.0.0</version>
       </dependency>
+      <dependency>
+        <groupId>io.grpc</groupId>
+        <artifactId>grpc-context</artifactId>
+        <version>1.12.0</version>
```
Comment (honnix) [inline pom.xml]: Internally we use 1.4.0 due to compatibility issue. If 1.4.0 can work in this case, we might use that.

File: flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Override
+  public <T> Value<T> invokeProcessFn(TaskId taskId, Fn<Value<T>> processFn) {
+    try {
+      return Context.current()
+          .withValue(TASK_ID, taskId.toString())
+          .withValue(TASK_NAME, taskId.name())
+          .withValue(TASK_ARGS, taskId.args())
+          .call(() -> super.invokeProcessFn(taskId, processFn));
+    } catch (Exception e) {
```
Comment (honnix) [inline flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java]: Will it look better having two catches?

File: flo-workflow/src/main/java/com/spotify/flo/Tracing.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.spotify.flo;
+
+import io.grpc.Context;
+
+public class Tracing {
```
Comment (honnix) [inline flo-workflow/src/main/java/com/spotify/flo/Tracing.java]: This class needs a bit more touch, static, final, private constructor, etc.

File: flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import static com.spotify.flo.Tracing.TASK_ID;
+
+import com.spotify.flo.EvalContext;
+import com.spotify.flo.Fn;
+import com.spotify.flo.Task;
+import com.spotify.flo.TaskId;
+import io.grpc.Context;
+
+/**
+ * A {@link EvalContext} that sets the current {@link Task#id()} on the {@link Context} when calling the process fn.
```
Comment (NatashaL) [inline flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java]: Update description

**Comment by NatashaL** [comment]
👍

File: flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java (hunk 4 above)
Comment (danielnorberg) [reply flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java]: How does it need to be updated?

Comment (NatashaL) [reply flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java]: Oh my bad, I read EvalContext != TracingContext and assumed it was a copy-paster

Comment (honnix) [reply flo-workflow/src/main/java/com/spotify/flo/context/TracingContext.java]: I would probably do the same. xD

**Pull request merged**